glom/core.py

```diff
@@ -100,7 +107,42 @@ class GlomError(Exception):
     (e.g., ``len``, ``sum``, any ``lambda``) will not be wrapped in a
     GlomError.
     """
-    pass
+    @classmethod
+    def wrap(cls, exc):
+        class GlomWrapError(type(exc), GlomError): pass
+        return GlomWrapError(*exc.args)
+        # this is another pattern, but doesn't work for C-defined exceptions
+        # (maybe if hasattr(sys.modules[type(exc).__module__], "__file__")?)
+        exc2 = copy.copy(exc)
+        exc2.__class__ = GlomWrapError
+        return exc2
```

Comment:
Is this cleaner in any way? Probably better to have just one way, right?

Reply:
nope, my exception-fu is a little weak here

Reply:
I'm thrashing around a bit